src/Compilers/Core/Portable/Operations/ControlFlowGraph.cs

```diff
@@ -73,51 +82,63 @@ public sealed partial class ControlFlowGraph
             //                      However, we should consider if we want to perform our own validation for the input.
             //                      In any case, we need to add unit tests to cover behavior of this API with respect to
             //                      an invalid input (including null).
 
-            (ControlFlowRegion enclosing, IOperation operation, int ordinal) = _methodsMap[nestedMethod];
-            Debug.Assert(nestedMethod == NestedMethods[ordinal]);
+            (ControlFlowRegion enclosing, ILocalFunctionOperation operation, int ordinal) = _localFunctionsMap[localFunction];
+            Debug.Assert(localFunction == LocalFunctions[ordinal]);
 
-            if (_lazySubGraphs == null)
+            if (_lazyLocalFunctionsGraphs == null)
             {
-                Interlocked.CompareExchange(ref _lazySubGraphs, new ControlFlowGraph[NestedMethods.Length], null);
+                Interlocked.CompareExchange(ref _lazyLocalFunctionsGraphs, new ControlFlowGraph[LocalFunctions.Length], null);
             }
 
-            if (_lazySubGraphs[ordinal] == null)
+            if (_lazyLocalFunctionsGraphs[ordinal] == null)
             {
-                ControlFlowGraph graph;
-
-                switch (operation.Kind)
-                {
-                    case OperationKind.LocalFunction:
-                        Debug.Assert(nestedMethod == ((ILocalFunctionOperation)operation).Symbol);
-                        graph = ControlFlowGraphBuilder.Create(operation, enclosing);
-                        break;
-                    
-                    case OperationKind.AnonymousFunction:
-                        Debug.Assert(nestedMethod == ((IAnonymousFunctionOperation)operation).Symbol);
-                        graph = ControlFlowGraphBuilder.Create(operation, enclosing);
-                        break;
-
-                    default:
-                        throw ExceptionUtilities.UnexpectedValue(operation.Kind);
-                }
-
-                Interlocked.CompareExchange(ref _lazySubGraphs[ordinal], graph, null);
+                Debug.Assert(localFunction == operation.Symbol);
+                ControlFlowGraph graph = ControlFlowGraphBuilder.Create(operation, enclosing, _captureIdDispenser);
+                Debug.Assert(graph.OriginalOperation == operation);
+                Interlocked.CompareExchange(ref _lazyLocalFunctionsGraphs[ordinal], graph, null);
             }
 
-            return _lazySubGraphs[ordinal];
+            return _lazyLocalFunctionsGraphs[ordinal];
+        }
+
+        /// <summary>
+        /// Creates a control flow graph for the given <paramref name="anonymousFunction"/>.
+        /// </summary>
+        public ControlFlowGraph GetAnonymousFunctionControlFlowGraph(IFlowAnonymousFunctionOperation anonymousFunction)
+        {
+            // PROTOTYPE(dataflow): It looks like the indexing below will throw right exceptions on all invalid inputs.
```

Comment:
Are you planning on following up on this immediately?

Reply:
> Are you planning on following up on this immediately?

Someone will, not necessarily me

---
In reply to: [193832740](https://github.com/dotnet/roslyn/pull/27536#discussion_r193832740) [](ancestors = 193832740)